Sources/Sentry/include/SentrySDK.h

```diff
@@ -58,6 +58,12 @@ SENTRY_NO_INIT
  */
 + (void)startWithOptionsObject:(SentryOptions *)options NS_SWIFT_NAME(start(options:));
 
++ (void)startWithDsn:(NSString *)dsn NS_SWIFT_NAME(start(dsn:));
+
++ (void)startWithDsn:(NSString *)dsn
+    withConfigureOptionsBlock:(void (^)(SentryOptions *options))configureOptions
+    NS_SWIFT_NAME(start(dsn:configureOptionsBlock:));
```

Comment:
```suggestion
+ (void)startConfigureOptionsBlock:(void (^)(SentryOptions *options))configureOptions
    NS_SWIFT_NAME(start(configureOptionsBlock:));
```
See comment above

Reply:
So would you prefer @HazAT and @marandaneto?
```
SentrySDK.start { o in
    o.dsn = "DSN"
    o.beforeSend = { event in
        return event
    }
   // ...
}
```

Reply:
The current problem with the above solution is that `SentryOptions.dsn` is a property of type `SentryDsn`, which is a private class. So we either need to make it public, but then you have to init the DSN yourself:
```
SentrySDK.start { o in
    do { 
         o.dsn = try SentryDsn("dsn")
    } catch {
        // ...
    }
    o.beforeSend = { event in
        return event
    }
   // ...
}
```
Which might not be the nicest first interaction with our SDK and we don't have the logic in here that the SDK gets disabled when the dsn is empty.

Reply:
also on Android, this is how it should be IMO

Reply:
@philipphofmann this problem is addressed on Android, I'd suggest just doing the same.
`String dsn` on options, as its not a breaking change, I'm fine with that.

and then on Sentry.start(...) while processing the options, you do:

`Dsn parsedDsn = new Dsn(dsn);`

if `dsn` is not valid, you do `throw new IllegalArgumentException("Invalid DSN: No public key provided.");`

In this case, App should crash, user should now it's not valid and they should fix right away.
if DSN is empty, SDK is disabled.
if DSN is null or not valid, IllegalArgumentException

Reply:
That said, adding the typed `SentryDsn` in the public API like in .NET was a mistake.
Ideally we take only the string and try to parse internally (and then use the typed DSN). Exposing this has proven to make the API more complex with very little added value.

No one is using the SentryDsn to parse a string and access the segments of the DSN like key and project id, or at least that never came up afaik.

Reply:
I agree with @bruno-garcia here, that exposing `SentryDsn` makes the API more complex and doesn't add any value.

So we still could go with the option of changing `SentryOptions.dsn` to `NSString` and not exposing `SentryDsn`, but then we have still need to solve the issue of validating the dsn and communicating this appropriate to the user.

Reply:
on Android, we don't expose the Dsn object, and we do the validation and we do the communication if it's invalid.
so If I get it right, my suggestion above is a good one or objections? I'm a bit lost here.